controllers/nodemaintenance_controller.go

```diff
@@ -159,7 +165,16 @@ func (r *NodeMaintenanceReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	r.logger.Info("Applying maintenance mode", "node", nodeName, "reason", nm.Spec.Reason)
 	node, err := r.fetchNode(ctx, drainer, nodeName)
 	if err != nil {
-		return r.onReconcileError(ctx, nm, drainer, err)
+		if apiErrors.IsNotFound(err) {
+			r.logger.Error(err, "Didn't find a node matching the NodeName field", "NodeName", nodeName)
+			// maintenance has failed - nodeName doesn't match an existing node
+			utils.WarningEvent(r.Recorder, nm, utils.EventReasonFailedMaintenance, utils.EventMessageFailedMaintenance)
+			nm.Status.Phase = v1beta1.MaintenanceFailed
+			return r.onReconcileError(ctx, nm, drainer, err)
```

Comment:
I'm not sure there is a point getting into reconcile error look when the node isn't found.
IIUC this error won't be resolved by future reconciles

Reply:
Good point, now it still uses the `onReconcileError` function but it won't return an error.